lib/plugins/aws/package/compile/events/cognito-user-pool.js

```diff
@@ -259,6 +315,35 @@ class AwsCompileCognitoUserPoolEvents {
     return null;
   }
 
+  checkCustomSenderConstraints(kmsKeyId, lambdaVersion, poolKmsIdMap, currentPoolName) {
+    if (lambdaVersion && !validLambdaVersions.includes(lambdaVersion)) {
+      throw new ServerlessError(
+        `Invalid Lambda version. Valid Lambda versions: ${validLambdaVersions}.`,
+        'COGNITO_INVALID_LAMBDA_VERSION'
+      );
+    }
+
+    if (!kmsKeyId) {
```

Comment:
Shouldn't it also be enforced on the config schema validation level?

Reply:
This is the only check that I am not sure about. I did some research in how validation is handled and I am not sure how to require `kmsKeyId` only with `CustomSMSSender` and `CustomEmailSender`, but leave it optional for all the other triggers. The attribute (like the `LambdaVersion` attribute) is marked as optional in AWS docs, but is required when using  `CustomSMSSender` and `CustomEmailSender`.

See image:
![image](https://user-images.githubusercontent.com/11302370/178059874-b4351f56-2aa4-4545-bbd3-be14529799b0.png)

Reply:
Further, `kmsKeyId` is only used with `CustomSMSSender` and/or `CustomEmailSender` according to [AWS Docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cognito-userpool-lambdaconfig.html#cfn-cognito-userpool-lambdaconfig-kmskeyid).

So, `kmsKeyId` is either not used (not even optional) or required (only with `CustomSMSSender` and/or `CustomEmailSender`).